docs/clients/grpc/subscriptions.md

```diff
@@ -152,58 +152,79 @@ There are a number of events in EventStoreDB called system events. These are pre
 
 This will only subscribe to events with a type that begin with `customer-`.
 
 #### Filtering by regular expression
 
-If you want to subscribe to multiple event types then it might be better to provide a regular expression.
+It might be advantageous to provide a regular expression when you want to subscribe to multiple event types.
 
 @[code{event-type-regex}](@grpc:server_side_filtering.py;server-side-filtering.js;server-side-filtering.ts;server_side_filtering/ServerSideFiltering.java;server-side-filtering/Program.cs;serverSideFiltering.go;server_side_filtering.rs)
 
 This will subscribe to any event that begins with `user` or `company`.
 
 ### Filtering by stream name
 
-If you only want to subscribe to a stream with a given name there are two options. You can either use a regular expression or a prefix.
+If you want to subscribe to a stream only with a given name, you have two options: you can use a regular expression or a prefix.
 
 #### Filtering by prefix
 
-If you want to filter by prefix pass in a `SubscriptionFilterOptions` to the subscription with an `StreamFilter.Prefix`.
+If you want to filter by prefix, pass in a `SubscriptionFilterOptions` to the subscription with an `StreamFilter.Prefix`.
 
 @[code{stream-prefix}](@grpc:server_side_filtering.py;server-side-filtering.js;server-side-filtering.ts;server_side_filtering/ServerSideFiltering.java;server-side-filtering/Program.cs;serverSideFiltering.go;server_side_filtering.rs)
 
-This will only subscribe to all streams with a name that begin with `user-`.
+This will only subscribe to all streams with a name that begins with `user-`.
 
 #### Filtering by regular expression
 
-If you want to subscribe to multiple streams then it might be better to provide a regular expression.
+If you want to subscribe to multiple streams, providing a regular expression might be better.
 
 @[code{stream-regex}](@grpc:server_side_filtering.py;server-side-filtering.js;server-side-filtering.ts;server_side_filtering/ServerSideFiltering.java;server-side-filtering/Program.cs;serverSideFiltering.go;server_side_filtering.rs)
 
 This will subscribe to any stream with a name that begins with `account` or `savings`.
 
 ## Checkpointing
 
-There is one thing to consider with server-side filtering, and that is when events that match your filter are few and far between. In this scenario, you might find yourself in the situation where EventStoreDB has searched through 1 million events, and the last thing you want to happen is for the server to get to event 900k and then have your client crash. It won't have been able to take a checkpoint and upon a restart, you'd have to go back to the beginning and start again.
+When a catch-up subscription is used to process an `$all` stream containing many events, the last thing you want is for your application to crash midway, forcing you to restart from the beginning.
```

Comment:
```suggestion
When processing an `$all` stream with a catch-up subscription, ensure your application stays stable to avoid restarting from the beginning.
```